### PR TITLE
cpu/atmega_common: adjust spi for at328p and at1281

### PR DESCRIPTION
**Pull request by dnahm**
This PR fixes the SPI pin configuration for ATMEGA328P and adds a SPI pin configuration for ATMEGA1281.

Because the existing test program "tests/periph_spi" needs more than 2KB RAM, it cannot be used on the ATMEGA328P (Arduino UNO). Therefore, a thin version of the test program was created. It can be found [here](https://github.com/dnahm/RIOT/tree/test_spi_write_read_function)

### CODE REVIEW

File: boards/waspmote-pro/Makefile.features [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # Put defined MCU peripherals here (in alphabetical order)
+FEATURES_PROVIDED += periph_adc
```
Comment (smlng) [inline boards/waspmote-pro/Makefile.features]: unrelated?

**Comment by PeterKietzmann** [comment]
@kYc0o when I saw the code I was pretty confused about the documentation of pins in the actual driver implementation. Is it intentional?

**Comment by kYc0o** [comment]
> @kYc0o when I saw the code I was pretty confused about the documentation of pins in the actual driver implementation. Is it intentional?

What do you mean with that? AFAIK, the port configuration was done according to the datasheet of atmega1281 MCU, on which it states that:

```
PB0 -> CS (hardware)
PB1 -> CLK
PB2 -> MOSI
PB3 -> MISO
```

Which as far as I can see correspond to the documentation. Have I missed something?

Regarding atmega328p I think this is only a mistake on the review, since the pins correspond to those on the atmega1281. Just a copy/paste I think. The port for atmega328p was done by someone else btw, but I think I reviewed the code.

**Comment by PeterKietzmann** [comment]
I was wondering about the location of the pin documentation. I have assumed it in a `periph_conf.h` file or so, but not in the actual driver which is actually shared for different boards **and** MCUs

**Comment by mali** [comment]
> Regarding atmega328p I think this is only a mistake on the review, since the pins correspond to those on the atmega1281. Just a copy/paste I think.
Certainly me ...

**Comment by kYc0o** [comment]
Yeah you're right @PeterKietzmann, I think the doc should be moved to periph_conf.h. Can @dnahm do it in this PR?

File: cpu/atmega_common/periph/spi.c [hunk 2]
```diff
@@ -47,17 +49,28 @@ void spi_init(spi_t bus)
...
      * master correctly, though we do not use it for now (as we handle the chip
      * select externally for now)
      */
-    DDRB |= ((1 << DDB2) | (1 << DDB1) | (1 << DDB0));
+     #if defined (CPU_ATMEGA2560) || defined (CPU_ATMEGA1281)
+     DDRB |= ((1 << DDB2) | (1 << DDB1) | (1 << DDB0));
+     #endif
+     #ifdef CPU_ATMEGA328P
+     DDRB |= ((1 << DDB2) | (1 << DDB3) | (1 << DDB5));
+     #endif
```
Comment (kYc0o) [inline cpu/atmega_common/periph/spi.c]: BTW indentation on the `#ifdef` should be removed.

**Comment by dnahm** [comment]
**@kYc0o** Yes, of course. I will move the SPI pin documentation to periph_conf.h

**Comment by dnahm** [comment]
**@kYc0o** OK to squash?

**Comment by kYc0o** [comment]
Yes, however I'm not able to test until tomorrow afternoon...

File: cpu/atmega_common/periph/spi.c (hunk 2 above)
Comment (kYc0o) [inline cpu/atmega_common/periph/spi.c]: Did you change this indentation? In my browser it appears without change but maybe is only my browser.

**Comment by dnahm** [comment]
**@kYc0o** Hi. Have you tested the PR? Is it OK to be merged?

**Comment by kYc0o** [comment]
I'm about to test but I'd like to have your example in... Can you PR it?

**Comment by miri64** [comment]
Did the testing happen?

**Comment by dnahm** [comment]
**@kYc0o**  My example is [here](https://github.com/dnahm/RIOT/tree/test_spi_write_read_function). It is a small version of "test/periph_spi".

**Comment by kYc0o** [comment]
I know that the example is there, what I mean is that is better if you make a PR with it, so the CI can build it for the UNO board.

Actually, there is a flaw in our CI since avr-gcc does not return any error when linking binaries larger than the actual available memory on the device, as you can see [here](https://ci.riot-os.org/RIOT-OS/RIOT/7225/2dae2c2910a4469d381de012c7434f8cf904a338/output/compile/tests/periph_spi/arduino-uno.txt). It's obvious it doesn't fit but the build is "correct". Thus, the UNO board should be blacklisted for this example (and so many others).

So what I propose, is that, either you add your valid test in this PR or in another one on top of this one, and you whitelist arduino-uno and duemilanove. Thi way we have at least one test which can work on the actual hardware.

**Comment by PeterKietzmann** [comment]
This PR is independent from any test. It rather fixes a bug on atmega328-based boards. The common SPI driver persuades to run on all atmega boards which is not the case without this PR. Whitelisting or blacklisting AVR boards is beyond this PR. A minimal SPI test application as provided by @dnahm is very useful indeed and may follow in future. Such app would be beneficial for several boards, no need for a whitelist though. We are short before release sprint and this is a bug fix. Let's merge it before testing begins.

**Comment by kYc0o** [comment]
Maybe I'm missing something but I can't see any bug and/or fix here... anyways, I tested it in a waspmote-pro and arduino-uno and it works. So go!

File: cpu/atmega_common/periph/spi.c (hunk 2 above)
Comment (kYc0o) [inline cpu/atmega_common/periph/spi.c]: I still don't see the indentation fixed, I thought it was my browser but I checked out the code and the indent is 4 spaces. Can you re-fix that before merging?

**Comment by dnahm** [comment]
**@kYc0o** The indentation in "cpu/atmega_common/periph/spi.c" is fixed.

**Comment by PeterKietzmann** [comment]
Great! I just checked the atmega-2560 again, everything works as expected. As fas as I see it, @dnahm directly addressed your comment about indentation. ACK and go
